src/main/java/reposense/parser/ArgsParser.java

```diff
@@ -203,15 +209,54 @@ public static CliArguments parse(String[] args) throws HelpScreenException, Pars
         }
     }
 
+    /**
+     * Returns a since date that is one month before {@code cliUntilDate} (if present) or one month before report
+     *  generation date otherwise.
+     */
+    private static Date getSinceDate(Optional<Date> cliUntilDate) {
```

Comment:
Rename this function to something it does. i.e. getDateAMonthBefore/Earlier/MinusAMonth?

Reason: You got too many `getSinceDate` references.

Also, extra space within  `*  generation`.